mindsdb/utilities/log.py

```diff
@@ -1,94 +1,77 @@
+"""
+This module sets up logging when called in different threads.
+
+Each thread that imports this module will automatically create the top level mindsdb logger and configure it according
+to the config provided by Config(). Presumably Config() will provide the local config.
+
+The module also sets up telemetry.
+
+Finally, calling initialize_log will create a child logger for the caller which will be used to create children with the
+get log function. This allows the thread to establish a unique child logger.
+
+Absent a call to initialize_log, the default logger will be the root mindsdb logger.
+
+"""
+
 import os
-import sys
 import logging
 
 from mindsdb.utilities.config import Config
 from functools import partial
 
+config = Config().get_all()
 
-class LoggerWrapper(object):
-    def __init__(self, writer_arr, default_writer_pos):
-        self._writer_arr = writer_arr
-        self.default_writer_pos = default_writer_pos
+log = logging.getLogger('mindsdb')
+log.propagate = False
+log.setLevel(min(
+    getattr(logging, config['log']['level']['console']),
+    getattr(logging, config['log']['level']['file'])
+))
 
-    def write(self, message):
-        if len(message.strip(' \n')) == 0:
-            return
-        if 'DEBUG:' in message:
-            self._writer_arr[0](message)
-        elif 'INFO:' in message:
-            self._writer_arr[1](message)
-        elif 'WARNING:' in message:
-            self._writer_arr[2](message)
-        elif 'ERROR:' in message:
-            self._writer_arr[3](message)
-        else:
-            self._writer_arr[self.default_writer_pos](message)
+formatter = logging.Formatter('%(levelname)s: - %(asctime)s - %(name)s - %(message)s')
 
-    def flush(self):
-        pass
+console_handler = logging.StreamHandler()
+console_handler.setLevel(config['log']['level'].get('console', logging.INFO))
+console_handler.setFormatter(formatter)
 
-    def isatty(self):
-        return True  # assumes terminal attachment
+log.handlers.clear()
+log.addHandler(console_handler)
+log.info(f"Root logger set to loglevel {log.level}")
+log.info(f"Root handler set to loglevel {console_handler.level}.")
+log.info(f"Number of handlers: {len(log.handlers)}")
+log.info(f"")
 
-    def fileno(self):
-        return 1  # stdout
+log.error = partial(log.error, exc_info=True)
 
+logger = log
 
-# default logger
-logger = logging.getLogger('dummy')
+# activate telemetry
+telemtry_enabled = os.getenv('CHECK_FOR_UPDATES', '1').lower() not in ['0', 'false', 'False']
```

Comment:
Typo.  "telemetry" not "telemtry"  (missing the last e)

Reply:
I realize wasn't your fault, but lets fix bad typos up while we're making this change.  ;)